File: ios/sample/SearchSample/ResultRow.swift
/// A search result row: the matched record plus which of its fields matched.
struct ResultRow: Identifiable, Hashable {
    let record: Record
    let matchedSlots: [UInt8]
    /// Marked normalized text per matched slot, as produced by the engine's
    /// `highlight` (matches wrapped in `Highlight.open`/`Highlight.close`).
    /// A slot is present only when the query actually matched its field.
    let highlights: [UInt8: String]
    var id: Int64 { record.id }
}

File: android/sample/app/src/main/java/com/unfydqry/searchsample/ResultRow.kt
package com.unfydqry.searchsample

/// A search result row: the record plus which of its fields matched.
///
/// [highlights] holds the marked normalized text per matched slot, as produced
/// by the engine's `highlight` (matches wrapped in `Highlight.OPEN`/`CLOSE`).
/// A slot is present only when the query actually matched its field.
data class ResultRow(
    val record: Record,
    val matchedSlots: List<UByte>,
    val highlights: Map<Int, String> = emptyMap(),
)
